backend/django/core/views/api_annotate.py

```diff
@@ -427,29 +428,50 @@ def data_unlabeled_table(request, project_pk):
     Returns:
         data: a list of data information
     """
-    project = Project.objects.get(pk=project_pk)
+    unlabeled_data = get_unlabeled_data(project_pk)
+    data = []
+    for d in unlabeled_data:
+        if len(data) < 50:
+            serialized_data = DataSerializer(d, many=False).data
 
-    stuff_in_queue = DataQueue.objects.filter(queue__project=project)
-    queued_ids = [queued.data.id for queued in stuff_in_queue]
+            temp = {
+                "Text": serialized_data["text"],
+                "metadata": serialized_data["metadata"],
+                "ID": d.id,
+            }
+            data.append(temp)
 
-    recycle_ids = RecycleBin.objects.filter(data__project=project).values_list(
-        "data__pk", flat=True
-    )
-    unlabeled_data = (
-        project.data_set.filter(datalabel__isnull=True)
-        .exclude(id__in=queued_ids)
-        .exclude(id__in=recycle_ids)
-    )
+    return Response({"data": data})
+
+
+@api_view(["GET"])
+@permission_classes((IsAdminOrCreator,))
+def search_data_unlabeled_table(request, project_pk):
```

Comment:
Thanks for setting all of this up! I was looking at this and wanted to see if I could speed it up by using some of the built-in django operations and it seems to work. 

Basically, if you use the built-in "icontains" filter which is a case-insensitive text match and dump everything in the serializer together you can get the same output with the code below. I *think* this is faster? 

```
    unlabeled_data = get_unlabeled_data(project_pk)
    text = request.GET.get("text")
    unlabeled_data = unlabeled_data.filter(text__icontains=text.lower())
    serialized_data = DataSerializer(unlabeled_data, many=True).data
    data = [{
        "data": d["text"],
        "metadata": d["metadata"],
        "id": d["pk"],
        "project": project_pk,
    } for d in serialized_data]
    return Response({"data": data})
```